document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -74,6 +75,7 @@ public class MongoCollection implements Collection {
 
   private final com.mongodb.client.MongoCollection<BasicDBObject> collection;
   private final MongoQueryExecutor queryExecutor;
+  private final Clock clock;
```

Comment:
I think, we can directly use `System.currentTimeMillis()`.

Reply:
Preferring `Clock` instead of `System.currentTimeMillis()` because of the improved [testability](https://www.baeldung.com/java-override-system-time).